control-plane-operator/controllers/hostedcontrolplane/cco/reconcile.go

```diff
@@ -0,0 +1,454 @@
+				PriorityClass: config.DefaultPriorityClass,
+			},
+			SetDefaultSecurityContext: setDefaultSecurityContext,
+			ReadinessProbes: config.ReadinessProbes{
+				containerMain().Name: {
+					ProbeHandler: corev1.ProbeHandler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/metrics",
+							Port:   intstr.FromInt(metricsPort),
+							Scheme: corev1.URISchemeHTTPS,
+						},
+					},
+					InitialDelaySeconds: 15,
+					PeriodSeconds:       60,
+					SuccessThreshold:    1,
+					FailureThreshold:    3,
+					TimeoutSeconds:      5,
+				},
+			},
+			LivenessProbes: config.LivenessProbes{
+				containerMain().Name: {
+					ProbeHandler: corev1.ProbeHandler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/metrics",
+							Port:   intstr.FromInt(metricsPort),
+							Scheme: corev1.URISchemeHTTPS,
+						},
+					},
+					InitialDelaySeconds: 60,
+					PeriodSeconds:       60,
+					SuccessThreshold:    1,
+					FailureThreshold:    5,
+					TimeoutSeconds:      5,
+				},
+			},
+			Resources: config.ResourcesSpec{
+				containerMain().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+						corev1.ResourceMemory: resource.MustParse("75Mi"),
+					},
+				},
+				containerMetrics().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("20m"),
+						corev1.ResourceMemory: resource.MustParse("10Mi"),
+					},
+				},
+				containerClientTokenMinter().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+						corev1.ResourceMemory: resource.MustParse("10Mi"),
+					},
+				},
+			},
+		},
+	}
+	params.deploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
+	if hcp.Annotations[hyperv1.ControlPlanePriorityClass] != "" {
+		params.deploymentConfig.Scheduling.PriorityClass = hcp.Annotations[hyperv1.ControlPlanePriorityClass]
+	}
+	params.deploymentConfig.SetDefaults(hcp, selectorLabels(), pointer.Int(1))
+	params.deploymentConfig.SetReleaseImageAnnotation(hcp.Spec.ReleaseImage)
+	return params
+}
+
+func ReconcileDeployment(deployment *appsv1.Deployment, params Params) error {
+	params.OwnerRef.ApplyTo(deployment)
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: selectorLabels(),
+		},
+		Strategy: appsv1.DeploymentStrategy{
+			Type: appsv1.RecreateDeploymentStrategyType,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: selectorLabels(),
+			},
+			Spec: corev1.PodSpec{
+				AutomountServiceAccountToken: pointer.Bool(false),
+				Containers: []corev1.Container{
+					util.BuildContainer(containerMain(), buildMainContainer(params.operatorImage, params.releaseVersion)),
+					util.BuildContainer(containerMetrics(), buildMetricsContainer(params.kubeRbacProxyImage)),
+					util.BuildContainer(containerClientTokenMinter(), buildClientTokenMinter(params.tokenMinterImage, params.issuerURL)),
```

Comment:
what's the purpose of buildClientTokenMinter here?

Reply:
The same as it is elsewhere in this code-base - we mint tokens for the CCO ServiceAccount on the tenant cluster for this component to use.

Reply:
Take ingress as an example, we sync down the secret https://github.com/openshift/hypershift/blob/c9c81140bd3250ffe23241a9adf24bbcbd0b4814/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L1155
with a format like
```
role_arn = %s
web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
```
Then the token minter generates a token in that path. The operator consumes the well known secret https://github.com/openshift/cluster-ingress-operator/blob/0cdc132376aab6f665e8f228bb07378357bb1e48/pkg/operator/controller/dns/controller.go#L610
And so the token minter enables the operator management side to access the AWS guest account. 

Then we make kubeconfig out of the SA available for the operator to talk to guest cluster https://github.com/openshift/hypershift/blob/c9c81140bd3250ffe23241a9adf24bbcbd0b4814/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L2926

I think CCO use case is not the above and in your case you are just minting the token as a way to give the operator guest cluster access through the existing SA, correct?

Reply:
For the simply access guest cluster use case is there a reason to not consolidate cco and registry with dns and ingress operators and use ReconcileServiceAccountKubeconfig for all instead of token-minter @sjenning @csrwng?

Reply:
`ReconcileServiceAccountKubeconfig` should be fine too, with explicit mounting

Reply:
Yep, seems like `pki.ReconcileServiceAccountKubeconfig` and manually mounting that in would work great for us.